### PR TITLE
Detect and classify lesions in the ISIC 2017 dataset using Mask R-CNN (s4528646)

### PR DESCRIPTION
**Pull request by joshua-morris**
I trained a Mask R-CNN in PyTorch on the ISIC 2017 lesion data set. The model is defined in `modules.py`, training is performed by `train.py` and an example prediction on the validation data set can be found in `predict.py`. The model achieves an average IoU of 0.75 on the validation set and better results may be possible applying data augmentations before training. The model was pretrained on COCO and then trained according to the methodology in the Readme

### CODE REVIEW

**Comment by nathasha-naranpanawa** [comment]
- Commit messages are OK
- Readme can be more descriptive
- Plots presented for loss
- Performance is OK. Almost 0.8 IoU achieved. 
- No classification accuracy shown
- Seems like no hyperparameter tuning performed to increase performnace of the model

**Comment by shakes76** [comment]
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate use and implementation, could be tuned -1
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem (no classification accuracy) -2
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage 
Module present
Commenting
No Data leakage
Difficulty: Medium -5
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe OK could be more descriptive -1
Good Description and Comments 
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description OK could be more descriptive -1
